src/llmq/quorums_chainlocks.cpp

```diff
@@ -407,6 +413,10 @@ bool CChainLocksHandler::IsTxSafeForMining(const uint256& txid)
 // This should also not be called from validation signals, as this might result in recursive calls
 void CChainLocksHandler::EnforceBestChainLock()
 {
+    if (!masternodeSync.IsBlockchainSynced()) {
```

Comment:
This might actually result in never reaching the correct chain and thus never leaving the `!masternodeSync.IsBlockchainSynced()` state. Very unlikely, but still possible.

Reply:
Hmm.. agree, will drop this one.